### PR TITLE
/fe: Move to svelte/kit

### LINKED ISSUE #259: Move to SvelteKit
We need to move the frontend to ESM to be able to install the latest Svizzles, and solve [this kind of issue](https://github.com/nestauk/dapsboard/issues/257) and use Kit to enhance DX.

### PR DESCRIPTION
**Pull request by saabi**
closes #259

### CODE REVIEW

File: fe/jsconfig.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+		"checkJs": false,
+		"esModuleInterop": true,
+		"forceConsistentCasingInFileNames": true,
+		"resolveJsonModule": true,
+		"skipLibCheck": true,
+		"sourceMap": true,
+		"strict": true,
+		"types": [
+			"@modyfi/vite-plugin-yaml/modules"
+			]
```
Comment (mindrones) [inline fe/jsconfig.json]: I'd remove the extra tab here

File: fe/netlify.toml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+[build]
+	# don't provide `base`
+	command = "npm run build"
+	environment = { NODE_VERSION = "18.13.0" }
+	publish = ".svelte-kit/netlify"
+
+[functions]
+	# temporary workaround for https://github.com/sveltejs/kit/issues/6462
+	node_bundler = "esbuild"
```
Comment (mindrones) [inline fe/netlify.toml]: needed? issue seems to be fixed

File: fe/package.json [hunk 3]
```diff
@@ -1,41 +1,31 @@
...
-		"compression": "^1.7.4",
 		"just-compare": "^2.3.0",
 		"lamb": "^0.61.0",
-		"polka": "^0.5.2",
-		"sirv": "^2.0.2",
 		"svelte-json-tree": "^1.0.0",
-		"xstate": "^4.14.0"
+		"xstate": "^4.35.2"
 	},
-	"description": "Dapsboard Front-end",
```
Comment (mindrones) [inline fe/package.json:13]: I'd reinstate `description`

File: fe/src/app.html [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+<head>
+	<meta charset='utf-8'>
+	<meta name='viewport' content='width=device-width, initial-scale=1.0'>
+	<meta name='theme-color' content='#333333'>
+	<link rel='stylesheet' href='%sveltekit.assets%/global.css'>
+	<link rel='manifest' href='%sveltekit.assets%/manifest.json'>
+	<link rel='icon' href='%sveltekit.assets%/favicon.png' />
+	%sveltekit.head%
+	</head>
+	<body data-sveltekit-preload-data='hover'>
```
Comment (mindrones) [inline fe/src/app.html]: `data-sveltekit-preload-data='hover'` needed?

File: fe/src/app.html [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+		<div id='app' style='display: contents'>%sveltekit.body%</div>
```
Comment (mindrones) [inline fe/src/app.html]: `style='display: contents'` needed?

File: fe/src/app.d.ts [hunk 6]
```diff
@@ -0,0 +1,13 @@
+// See https://kit.svelte.dev/docs/types#app
```
Comment (mindrones) [inline fe/src/app.d.ts]: do we need TS stuff?

File: fe/src/bin/make_data.js [hunk 7]
```diff
@@ -5,32 +5,35 @@
...
-const makeProject = string => getPathName(string).split('_')[0];
-const makeSource = string => getPathName(string).split('_')[1];
-const makeVersion = string => getPathName(string).split('_')[2].replace('v', '');
-const resolveSchema = string => path.resolve(INDICES_SPECS_DIR, string);
+const makeDatasetName = (/** @type {string} */ string) => getPathName(string).split('_v')[0];
+const makeLabel = (/** @type {string} */ string) => getPathName(string).replace(/_/gu, ' ');
+const makeProject = (/** @type {string} */ string) => getPathName(string).split('_')[0];
+const makeSource = (/** @type {string} */ string) => getPathName(string).split('_')[1];
+const makeVersion = (/** @type {string} */ string) => getPathName(string).split('_')[2].replace('v', '');
+const resolveSchema = (/** @type {string} */ string) => path.resolve(INDICES_SPECS_DIR, string);
```
Comment (mindrones) [inline fe/src/bin/make_data.js]: (1) can we remove this type annotations?

File: fe/src/bin/make_data.js [hunk 8]
```diff
@@ -42,14 +45,16 @@ const makeSchemaObj = applyFnMap({
 	version: makeVersion
 });
 
-const save = (data, dest) => saveObj(dest, 2)(data).then(tapMessage(`Saved ${dest}`));
+const save = (/** @type {{ spec: unknown; constructor: Function; toString(): string; toLocaleString(): string; valueOf(): Object; hasOwnProperty(v: PropertyKey): boolean; isPrototypeOf(v: Object): boolean; propertyIsEnumerable(v: PropertyKey): boolean; }[]} */ data, /** @type {string} */ dest) => saveObj(dest, 2)(data).then(tapMessage(`Saved ${dest}`));
```
Comment (mindrones) [inline fe/src/bin/make_data.js]: (1)

File: fe/src/bin/make_data.js [hunk 9]
```diff
@@ -42,14 +45,16 @@ const makeSchemaObj = applyFnMap({
...
 
 const run = async () => {
 	const refs = await readDir(INDICES_SPECS_DIR)
+	// @ts-ignore
```
Comment (mindrones) [inline fe/src/bin/make_data.js]: (1)

File: fe/src/lib/env.js [hunk 10]
```diff
@@ -0,0 +1,2 @@
+// @ts-ignore
```
Comment (mindrones) [inline fe/src/lib/env.js]: (1)

File: fe/src/routes/explore/+page.js [hunk 11]
```diff
@@ -0,0 +1,9 @@
+import * as _ from 'lamb';
+
+export function load (event) {
+    const {url: {searchParams}} = event;
+    const query = _.fromPairs(Array.from(searchParams.entries()));
+    const {project, source, version} = query;
+
+    return {project, source, version};
```
Comment (mindrones) [inline fe/src/routes/explore/+page.js]: use tabs (make sure this is true everywhere)

File: fe/src/routes/explore/[source]/+page.js [hunk 12]
```diff
@@ -0,0 +1,9 @@
+import * as _ from 'lamb';
+
+export function load (event) {
+    const {url: {searchParams}} = event;
+    const query = _.fromPairs(Array.from(searchParams.entries()));
```
Comment (mindrones) [inline fe/src/routes/explore/[source]/+page.js]: seems we could use a util to do this (potenially good for `@svizzle/utils`), say:

```js
const query = collectionToObject(searchParams)
```

File: fe/src/lib/utils/generic.js [hunk 13]
```diff
@@ -1,5 +1,6 @@
 import * as _ from 'lamb';
 import {getId, isFunction} from '@svizzle/utils';
+import {isClientSide} from '@svizzle/ui/src/utils/env.js';
```
Comment (mindrones) [inline fe/src/lib/utils/generic.js:3]: (2) Can we use `import { browser } from '$app/environment';`?

File: fe/src/routes/_datasets/[id]/+page.svelte [hunk 14]
```diff
@@ -15,7 +15,7 @@
 	import JSONTree from 'svelte-json-tree';
 
 	import {constructQuery} from '$lib/elasticsearch/utils/aggQuery';
-	import {IS_BROWSER} from 'utils/generic';
+	import {isClientSide} from '@svizzle/ui';
```
Comment (mindrones) [inline fe/src/routes/_datasets/[id]/+page.svelte]: (2)

**Review by mindrones — COMMENTED** [review]
pass 1

File: fe/src/app.html (hunk 4 above)
Comment (saabi) [reply fe/src/app.html]: Currently effectively working to preload resources on hover

File: fe/src/app.html (hunk 5 above)
Comment (saabi) [reply fe/src/app.html]: https://github.com/sveltejs/kit/discussions/7585

**Review by mindrones — APPROVED** [review]
GTM!

**Pull request merged**